### PR TITLE
Layers: Added serialization/ deserialization

### PR DESCRIPTION
**Pull request by Mugen87**
see https://github.com/mrdoob/three.js/issues/5562#issuecomment-397788029

### CODE REVIEW

File: src/core/Object3D.js [hunk 1]
```diff
@@ -663,6 +663,13 @@ Object3D.prototype = Object.assign( Object.create( EventDispatcher.prototype ),
 
 		if ( this.matrixAutoUpdate === false ) object.matrixAutoUpdate = false;
 
+		if ( this.layers.mask !== 1 ) {
+
+			object.layers = {};
```
Comment (mrdoob) [inline src/core/Object3D.js]: One-liner seems simpler?

```js
object.layers = { mask: this.layers.mask };
```

Comment (Mugen87) [reply src/core/Object3D.js]: 👍

**Comment by Mugen87** [comment]
👍

File: src/core/Object3D.js [hunk 2]
```diff
@@ -659,6 +659,7 @@ Object3D.prototype = Object.assign( Object.create( EventDispatcher.prototype ),
 		if ( this.renderOrder !== 0 ) object.renderOrder = this.renderOrder;
 		if ( JSON.stringify( this.userData ) !== '{}' ) object.userData = this.userData;
 
+		object.layers = { mask: this.layers.mask };
```
Comment (mrdoob) [inline src/core/Object3D.js]: Actually, considering how stable that class is, I think I would just do this:

```js
object.layers = this.layers.mask;
```

Comment (Mugen87) [reply src/core/Object3D.js]: Done.

**Comment by mrdoob** [comment]
Nice and simple 👌

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**